Resolved the majority of Warnings.

RoiEXLab opened the pull request:
Also removed unused imports.
Seems to work for me, maybe someone could double check if these changes work
like they should.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/triplea-game/triplea/569)
<!-- Reviewable:end -->

DanVanAtta commented on src/games/strategy/triplea/ai/fastAI/FastOddsEstimator.java:
```diff
@@ -16,14 +16,19 @@
-  private PlayerID attacker = null;
-  private PlayerID defender = null;
+  @SuppressWarnings("unused")
+private GameData data;
```
formatting, the formatter is in: https://github.com/triplea-game/triplea/tree/master/eclipse/format

DanVanAtta commented on src/games/strategy/triplea/ai/proAI/ProBidAI.java:
```diff
@@ -387,16 +387,19 @@ public void bidPlace(final IAbstractPlaceDelegate placeDelegate, final GameData
     }
-    final BattleDelegate delegate = DelegateFinder.battleDelegate(data);
+    @SuppressWarnings("unused")
+	final BattleDelegate delegate = DelegateFinder.battleDelegate(data);
```
pls use spaces, not tabs (formatter will take care of that)

DanVanAtta commented on src/games/strategy/triplea/ai/proAI/logging/ProLogUI.java:
```diff
@@ -10,7 +10,8 @@
  */
 public class ProLogUI {
-  private static TripleAFrame s_frame = null;
+  @SuppressWarnings("unused")
```
Hmm, I don't think just adding a suppress warning annotation is the right way to fix this. If the variable is not used, it can likely be removed. But be careful since some are used by reflection and are actually used.. Generally the stuff in remote and/or delegate code, or that has to do with XML parsing is potentially used by reflection. Testing though is the only way to know in the end, connect to lobby, play a game, and make sure you do not get a variable not found error.

DanVanAtta commented on src/games/strategy/util/PointFileReaderWriter.java:
```diff
@@ -125,17 +125,17 @@ private static void write(final StringBuilder buf, final OutputStream sink) thro
     final StringBuilder out = new StringBuilder();
     if (mapping == null) {
-      mapping = new HashMap();
+      mapping = new HashMap<String, Collection<Point>>();
```
Here you can use diamond notation, `new HashMap<>();`

DanVanAtta commented on src/games/strategy/util/PointFileReaderWriter.java:
```diff
@@ -149,6 +149,31 @@ public static void writeOneToMany(final OutputStream sink, Map mapping) throws E
     write(out, sink);
   }
+  
+  /*public static void writeOneToMany(final OutputStream sink, Map<String, List<Point>> mapping) throws Exception {
```
pls remove the commented out code

RoiEXLab replied:
Sorry, I tested, if java differs between Map<String, List<Point>> and Map<String, Collection<Point>> it does not...

DanVanAtta replied:
No worries, context is a bit immaterial here. IMO the best way to respond to PR review comments is by updating the code, let the code do the talking for itself. When you hit that point, it's a good sign that the code is maintainable. After this PR is closed, all the commentary in it will be rendered irrelevant. So placating me in this specific review is not the goal, the goal is to have it so that anyone reading this code will not have the same questions / comments.

DanVanAtta commented on src/util/image/DecorationPlacer.java:
```diff
@@ -452,7 +450,7 @@ private void saveImagePoints() {
       final FileOutputStream out = new FileOutputStream(fileName);
 
-      PointFileReaderWriter.writeOneToMany(out, m_currentPoints);
+      PointFileReaderWriter.writeOneToMany(out, new HashMap<String, Collection<Point>>(m_currentPoints));
```
can use diamond notation here as well

DanVanAtta commented on src/util/image/PlacementPicker.java:
```diff
@@ -481,7 +481,7 @@ private void savePlacements() {
       }
       final FileOutputStream out = new FileOutputStream(fileName);
-      PointFileReaderWriter.writeOneToMany(out, m_placements);
+      PointFileReaderWriter.writeOneToMany(out, new HashMap<String, Collection<Point>>(m_placements));
```
another opportunity for diamond notation.

DanVanAtta commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -3,4 +3,4 @@ distributionBase=GRADLE_USER_HOME
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
```
Could you provide some background on this change? Not clear why it was needed

RoiEXLab replied:
My eclipse changed this from all to bin... I just reversed this change

DanVanAtta commented:
Not a bad first PR - thank you! Please do look to removing unused variables rather than adding a suppress warning. I'd rather have the warning and know the variable can likely be removed rather than keeping it around with suppressed warning. When we see suppressed warning ideally they are for good reason, namely that it turns out the variable *is* actually used (via reflection, or some other manner that is not checked at compile time)

DanVanAtta commented:
I suppose a second point, we've been in the habit of trying to splice up PRs so they are small and independent, ideally with just one change in them. Makes review easier, and easier for parts of work rather than all at once.

I've a number of utility git functions you're welcome to check out: https://github.com/DanVanAtta/linux_tools/tree/master/bin

Of those, git_rebranch, git_squash are relatively useful. Otherwise to pull apart PRs I use "git rebase -i", then I move independent commits to the very top of the stack, this way I know if there any merge conflicts. Then 'git rebase --abort' in case anything went wrong. Also important to be sure you push your branch to remote first as well so you can get your work back in case the rebase is messed up. Then once I know which commits are independent, I'll remove them from the PR (using git rebase -i, simply delete the line with the commit), then will do a 'git_rebranch' to check out a new branch from master, then would use `git cherry-pick <commit SHA1> <commit SHA2> ..` to create teh new branch.

RoiEXLab commented:
Ok then I'll be focusing on resolving warnings.
There was just the thing, that I noticed that when Maps get loaded, it seems like you'd be able to close the window, but you are not. one line of code fixed this: `setUndecorated(true);`

DanVanAtta commented:
Please put your IP address in a GIST (https://gist.github.com/), and let us know the Gist URL in the gitter channel. We'll see about un-banning the IP.

The `setUndecorated(true)` change is very interesting. I do recommend removing that from this PR and putting just that one line in its own branch/commit/PR. Likely we could merge that very quickly when it is on its own and not get bogged down in the other details.

DanVanAtta commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,4 +1,4 @@
-#Sat Jan 09 14:28:41 CET 2016
+#Fri Apr 01 13:38:03 CEST 2016
```
This one can probably be reversed as well, no strict need to update it.

RoiEXLab replied:
This will always get updated if the file is accessed - a timestamp

DanVanAtta commented on src/games/strategy/net/UniversalPlugAndPlayHelper.java:
```diff
@@ -23,7 +23,8 @@
   private InternetGatewayDevice m_device = null;
-  private static UniversalPlugAndPlayHelper s_lastInstance = null;
+  @SuppressWarnings("unused")
+private static UniversalPlugAndPlayHelper s_lastInstance = null;
```
formatting still needs to be applied.
Also pls see about removing this variable too instead of suppressing the warning.

DanVanAtta commented on src/games/strategy/triplea/ai/fastAI/FastOddsEstimator.java:
```diff
@@ -16,13 +16,18 @@
 
 public class FastOddsEstimator implements IOddsCalculator {
 
+  @SuppressWarnings("unused")
```
Same, @SuppressWarnings is not the right fix, see about removing the variable.

DanVanAtta commented on src/util/image/DecorationPlacer.java:
```diff
@@ -614,8 +612,10 @@ private void fillCurrentImagePointsBasedOnTextFile(final boolean fillInAllTerrit
+      image = new File(
+          ClientFileSystemHelper.getRootFolder() + File.separator + ResourceLoader.RESOURCE_FOLDER + File.separator
+              + s_imagePointType.getFolderName(),
+          s_imagePointType.getImageName());
```
nit: formatting, line 618 should be combined with 617

DanVanAtta commented on src/util/image/PlacementPicker.java:
```diff
@@ -373,7 +373,7 @@ public void actionPerformed(final ActionEvent event) {
     menuBar.add(editMenu);
   }// end
-                                                                                                                                                                                                                                                                                                                                                                                                                ...
+   // constructor
```
Please do not add comments to indicate class structure. They've more a maintenance burden than anything else, very likely to become out dated and incorrect as the code changes.

RoiEXLab replied:
I have no idea where this comes from...

RoiEXLab commented:
Sorry... I had no time the last two days... I will proceed sometime soon...

DanVanAtta commented on src/games/strategy/util/PointFileReaderWriter.java:
```diff
@@ -125,19 +125,19 @@ private static void write(final StringBuilder buf, final OutputStream sink) thro
   }
 
-  public static void writeOneToMany(final OutputStream sink, Map mapping) throws Exception {
+  public static void writeOneToMany(final OutputStream sink, Map<String, Collection<Point>> mapping) throws Exception {
```
`Map<String,Collection<Point>>` likely would be simpler as a Guava BiMap.

RoiEXLab commented:
Accidently Updated from master in GitHub desktop...

DanVanAtta commented:
If you have access to command line, to undo the last commit:
```
$ git status   ## verify clean working state
$ git reset --soft HEAD^  ## this will move the git pointer back one commit
$ git push origin -f <name_of_your_branch> ## this will push your current branch to the remote, the "-f" forces the push, so anything you had before is overwritten. Which is good since the remote branch has an extra commit which the incoming commit does not have.
$ git reset .  ## This is just to fix your environment, so that the new committed files are no longer staged
$ git checkout .   ## This will now revert any of the staged files, which will finish the undo of a14d4f7
```

DanVanAtta commented:
Or, create a new branch, then cherry pick each commit:
`git cherry-pick  <commit_1> <commit_2>`

RoiEXLab commented:
Got it all reverted... GitHub Desktop seems buggy - when updating from master local files remain untouched and every attempt to sync them with github will fail.
Solution was to re-download the repository

I removed some of the unused variables... Seems all fine to me...
The only errors I get are due to the fact, having 2 versions of TripleA installed. e.g. Found too many map files etc.
Will proceed to get all the other unused Variables removed.

RoiEXLab commented:
An off-topic question... Is the buil.xml ant file still used? If not someone could remove it, because it fails when I try to use it.

RoiEXLab commented:
So I finished my work.
I'll leave the remaining warnings up to you, because I have no idea how the deprecated OSX API and the Territory distance calculation works.

DanVanAtta commented on gradle/wrapper/gradle-wrapper.properties:
```diff
@@ -1,6 +1,6 @@
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
```
Please add this file to .gitignore if it's difficult to keep it from changing. It should not be changed unless very intentional.

DanVanAtta replied:
This change still needs to be backed out :/

DanVanAtta commented on src/games/strategy/common/swing/SwingComponents.java:
```diff
@@ -143,13 +142,13 @@ public void windowDeactivated(WindowEvent e) {}
-  public static <T> JList<String> newJList(DefaultListModel listModel) {
-    return new JList(listModel);
+  public static <T> JList<String> newJList(DefaultListModel<String> listModel) {
+    return new JList<String>(listModel);
```
Diamond notation here pls

RoiEXLab replied:
Won't work, because on line 151 a new JList ist getting created.

DanVanAtta commented on src/games/strategy/engine/chat/ChatPlayerPanel.java:
```diff
@@ -128,13 +128,12 @@ private void setDynamicPreferredSize() {
+    m_players = new JList<>(m_listModel);
     m_players.setFocusable(false);
-    m_players.setCellRenderer(new ListCellRenderer() {
+    m_players.setCellRenderer(new ListCellRenderer<Object>() {
```
diamond notation here and again on line 209

RoiEXLab replied:
The Object is necessary here.
Same in 209.

DanVanAtta replied:
ok

DanVanAtta commented on src/games/strategy/engine/data/ChangeFactory.java:
```diff
@@ -105,6 +105,7 @@ public static Change changeResourcesChange(final PlayerID player, final Resource
   }
 
   /** TODO: unused code, kept per @veqryn */
+  @SuppressWarnings("unused")
```
remove the `@SuppressWarnings` unless this is actually used. Better to have the compiler error, otherwise what this annotationo is saying is that compile time checks are insufficient to show that this code *actually* is used (in which case if you give it the  axe, code will still compile, but bomb at runtime)

RoiEXLab replied:
I kept unused functions, because the involved logic was very complex and it would be a paint to reimplement all the stuff...

DanVanAtta replied:
You're assuming the requirements of the function will be identical in a  future scenario? Could you add the unit test for this code as well so then someone really could just uncomment it?

My point is, remove unused code. If someone really needs it, they'll find it in the history. Otherwise if someone really needs this functionality, chances are really good they'll re-implement it and not discover that this function was here anyways. In this example, I don't mind too much since there is a TODO to at least indicate it is not actually used. 

If we really want to keep code samples around, a wiki is actually a better place. At least there we won't have to maintain dead code when we refacgtor stuff.

DanVanAtta commented on src/games/strategy/engine/framework/map/download/DownloadFileParser.java:
```diff
@@ -26,6 +26,7 @@ private DownloadFileParser() {}
   }
 
   public static List<DownloadFileDescription> parse(final InputStream is) {
+    @SuppressWarnings("unchecked")
```
Same, please remove the `@SuppressWarnings`

RoiEXLab replied:
I added the @SuppressWarnings("unchecked") if there was no easy solution fixing the problem. It's simply not possible to check a generic type. Thats why i suppressed the warning - it's recommended to do so.

DanVanAtta replied:
In this case that is okay.

DanVanAtta commented on src/games/strategy/engine/framework/map/download/DownloadMapsWindow.java:
```diff
@@ -33,10 +33,8 @@
 import games.strategy.common.swing.SwingComponents;
 import games.strategy.util.Version;
 
-
```
There is going to be a merge conflict with this change depending upon who merges first.. FYI, no action to be taken.

DanVanAtta commented on src/games/strategy/engine/framework/map/download/DownloadRunnable.java:
```diff
@@ -84,6 +82,7 @@ private void downloadFile() {
     }
   }
 
+  @SuppressWarnings("unused")
```
Same, please remove the `@SuppressWarnings`

DanVanAtta commented on src/games/strategy/engine/framework/startup/ui/MetaSetupPanel.java:
```diff
@@ -287,6 +276,7 @@ private void connectToLobby() {
       return Optional.empty();
     }
     Yaml yaml = new Yaml();
+    @SuppressWarnings("unchecked")
```
Same, please remove the `@SuppressWarnings` unless the object is actually used (best solution is ot remove the unused object, failing that, leave the warning so we come back to it one day, hiding a problem is not a good thing)

DanVanAtta commented on src/games/strategy/engine/framework/startup/ui/editors/MicroWebPosterEditor.java:
```diff
@@ -163,10 +163,11 @@ private void initGame() {
     window.getContentPane().add(new JLabel("Select Players For Each Nation:"), new GridBagConstraints(0, 0, 2, 1, 0, 0,
         GridBagConstraints.CENTER, GridBagConstraints.NONE, new Insets(20, 20, 20, 20), 0, 0));
-    final JComboBox[] comboBoxes = new JComboBox[m_parties.length];
+    @SuppressWarnings("unchecked")
```
No need for the @SuppressWarnings, the TODO alone is fine, in which case we are getting rid of the compiler warning. Easier to find the compiler warning than it is a todo.

DanVanAtta commented on src/games/strategy/triplea/ai/AdvancedUtils.java:
```diff
@@ -82,6 +82,7 @@ public static Route trimRouteBeforeFirstTerWithEnemyUnits(final Route route, fin
   }
 
 
+  @SuppressWarnings("unchecked")
```
Same, please remove the `@SuppressWarnings`

DanVanAtta commented on src/games/strategy/triplea/ai/proAI/logging/ProLogWindow.java:
```diff
@@ -472,7 +472,7 @@ public void run() {
   private javax.swing.JCheckBox v_limitLogHistoryCB;
   private javax.swing.JSpinner v_limitLogHistoryToSpinner;
-  private javax.swing.JComboBox v_logDepth;
+  private javax.swing.JComboBox<String> v_logDepth;
```
Looks pretty good up to this point. Some   `@SuppressWarnings` were added that are going to bury a problem, rather than help. It's far better to have a compiler warning, than something that is actually unused, suppressed, and because it is  `@SuppressWarnings`, it'll look like it was intentionally left in because it *actually* is used, which is misleading. That leads us down a route of maintaining dead code, which is never fun to realize.

RoiEXLab commented:
To conclude: There were many situations in which a Generic Type got casted to another with a specific type. There is no solution to solve that unless the whole logic behind that gets rewritten.[Source](http://stackoverflow.com/questions/509076/how-do-i-address-unchecked-cast-warnings). In this Situations I added a SuppressWarnings("unchecked") and added a TODO the first couple times. I only kept unused methods, because I'm not sure if those will be needed sometime soon.

Notice, that there is a difference between "unused" and "unchecked" - I'm not quite sure if you noticed that.

RoiEXLab commented:
I updated the gitignore File and updated the Formatting in DecorationPlacer.java.
I'm not sure if I should remove the unused methods, keep them with supressed warnings comment the out or keep them with active warnings. What should I do in your opinion?

RoiEXLab commented:
Ok I removed the unused code snippets.
I also searched for some other supressed warnings and I found a complete unused class. Deleted it.

DanVanAtta commented:
Looks like just the gradle file needs to be reverted, otherwise looks pretty good. Please @RoiEXLab  do a double pass testing this, set up a local host game and connect to yourself locally. Some stuff is used by reflection, but it does not look like any of that was touched. The ClassLoader is a pre-java6 hack thing, I think we removed recently the last call into it, so not too surprising the entire class can get the axe.

DanVanAtta commented:
This PR should resolve the merge conflicts in this branch: https://github.com/RoiEXLab/triplea/issues/2

RoiEXLab commented:
I didn't see any merge conflicts before... Now there are some, because of your pull request.
Good or bad sign?

DanVanAtta commented:
@RoiEXLab  , not the best since I thought I had resolved them. The conflicts are from a a completely different PR that was merged last. That PR updated InstallMapDialog heavily, hence the conflicts. Are you able to resolve the conflicts @RoiEXLab ?

RoiEXLab commented:
I will look into that on Sunday evening or Monday... Lets see what I can do

RoiEXLab commented:
So I resolved the merge conflicts... Please merge this pull request as quickly as possible, so that I can stop worrying that I have to resolve errors again and again.

RoiEXLab commented:
A quick question: While resolving the warnings I noticed, that there is lots of code used, not matching the triplea formatting rules. Should I open a pull request on reformatting, all of the code when few other pull requests are open?

DanVanAtta commented:
- Sorry to say looks like we hit a conflict again already. 
- I see that the last 'blockers' have been addressed, this will be good to merge once the conflicts are settled

> "Should I open a pull request on reformatting, all of the code when few other pull requests are open?"
I would suggest not. Probably better for someone with write permission to coordinate a time and do it in one go.Which areas are you seeing that are missing formatting? Ideally it should not be too much. If it's less than 5 or 6 classes, I would say go ahead and do a reformatting PR. If larger, the potential for merge conflicts might make it not so worth it.

RoiEXLab commented:
Resolved merge conglicts again... Please hurry this time ^^
